### PR TITLE
feat: support http(s) proxy

### PR DESCRIPTION
**Pull request by fengmk2**
closes https://github.com/cnpm/npminstall/issues/135

### CODE REVIEW

**Comment by dead-horse** [comment]
CI 挂了，jshint 没过

**Comment by fengmk2** [comment]
先不合，有bug

**Comment by fengmk2** [comment]
https://www.npmjs.com/package/proxy-agent

File: lib/urllib.js [hunk 1]
```diff
@@ -113,6 +114,8 @@ var PROTO_RE = /^https?:\/\//i;
  *   - {Boolean} [timing]: Enable timing or not, default is `false`.
  *   - {Function} [lookup]: Custom DNS lookup function, default is `dns.lookup`.
  *       Require node >= 4.0.0 and only work on `http` protocol.
+ *   - {Boolean} [enableProxy]: optional, enable proxy request. Default is `false`.
+ *   - {String|Object} [proxy]: optional proxy agent uri or options. Default is `null`.
  * @param {Function} [callback], callback(error, data, res). If missing callback, will return a promise object.
```
Comment (atian25) [inline lib/urllib.js]: 逗号是上一行的吧？

**Comment by fengmk2** [comment]
2.24.0

**Pull request merged**